### PR TITLE
Bits of PhysicsBehavior

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

File: src/game/logic/object/update/physicsupdate.cpp [hunk 1]
```diff
@@ -108,3 +543,122 @@ bool PhysicsBehavior::Is_Motive() const
...
+
+    if (Is_Motive()) {
+        return UPDATE_SLEEP_TIME_MIN;
+    }
+
+    if (Get_Object()->Get_Layer() != LAYER_GROUND) {
+        return UPDATE_SLEEP_TIME_MIN;
+    }
+
+    if (Get_Object()->Is_Above_Terrain() || Get_Current_Overlap() || Get_Previous_Overlap() || !Get_Flag(UPDATE_EVER_RUN)) {
```
Comment (tomsons26) [inline src/game/logic/object/update/physicsupdate.cpp]: seperate this get object too then

File: src/game/logic/object/update/physicsupdate.cpp [hunk 2]
```diff
@@ -108,3 +543,122 @@ bool PhysicsBehavior::Is_Motive() const
...
+
+    if (!gm_isnanf(force->x) && !gm_isnanf(force->y) && !gm_isnanf(force->z)) {
+        float mass = Get_Mass();
+        Coord3D new_force = *force;
+
+        if (Is_Motive()) {
+            const Coord3D *dir = Get_Object()->Get_Unit_Dir_Vector2D();
+            float f = -dir->y * force->x + force->y * dir->x;
+            new_force.x = -dir->y * f;
+            new_force.y = f * dir->x;
```
Comment (tomsons26) [inline src/game/logic/object/update/physicsupdate.cpp]: flip the * si uts consistent

File: src/game/logic/object/update/physicsupdate.cpp [hunk 3]
```diff
@@ -108,3 +543,122 @@ bool PhysicsBehavior::Is_Motive() const
...
+    tm.Pre_Rotate_X(-roll);
+    tm.Pre_Rotate_Y(pitch);
+    tm.Pre_Rotate_Z(yaw);
+    Get_Object()->Set_Transform_Matrix(&tm);
+}
+
+void PhysicsBehavior::Do_Bounce_Sound(Coord3D const &pos)
+{
+    if (m_bounceSound != nullptr) {
+        AudioEventRTS sound(m_bounceSound->m_event);
```
Comment (tomsons26) [inline src/game/logic/object/update/physicsupdate.cpp:562]: missing code

File: src/game/logic/object/update/physicsupdate.cpp (hunk 2 above)
Comment (jonwil) [reply src/game/logic/object/update/physicsupdate.cpp]: fixed.

File: src/game/logic/object/update/physicsupdate.cpp (hunk 1 above)
Comment (jonwil) [reply src/game/logic/object/update/physicsupdate.cpp]: fixed.

File: src/game/logic/object/update/physicsupdate.cpp [hunk 4]
```diff
@@ -13,17 +13,452 @@
...
+    m_collided(OBJECT_UNK),
+    m_ignoreCollisionsWith(OBJECT_UNK),
+    m_projectileUpdateInterface(nullptr),
+    m_bounceSound(nullptr)
+{
+    m_accel.Zero();
+    m_prevAccel = m_accel;
+    m_vel.Zero();
+    Set_Allow_Bouncing(Get_Physics_Behavior_Module_Data()->m_allowBouncing);
+    Set_Allow_Collide_Force(Get_Physics_Behavior_Module_Data()->m_allowCollideForce);
```
Comment (tomsons26) [inline src/game/logic/object/update/physicsupdate.cpp:115]: missing setWakeFrame

File: src/game/logic/object/update/physicsupdate.cpp [hunk 5]
```diff
@@ -13,17 +13,452 @@
...
+                Vector3 v = tm.Get_X_Vector();
+                float angle = GameMath::Atan2(v.Z, GameMath::Sqrt(GameMath::Square(v.X) + GameMath::Square(v.Y)));
+
+                if (offset > 0.0f) {
+                    angle = DEG_TO_RADF(90) - angle;
+                } else {
+                    angle = angle + DEG_TO_RADF(-90);
+                }
+
+                pitch = pitch * angle;
```
Comment (tomsons26) [inline src/game/logic/object/update/physicsupdate.cpp]: Sin missing

File: src/game/logic/object/update/physicsupdate.cpp (hunk 4 above)
Comment (jonwil) [reply src/game/logic/object/update/physicsupdate.cpp:115]: fixed.

File: src/game/logic/object/update/physicsupdate.cpp [hunk 6]
```diff
@@ -13,17 +13,452 @@
...
+
+        if (obj->Get_Disabled_State(DISABLED_TYPE_DISABLED_FREEFALL)) {
+            obj->Clear_Disabled(DISABLED_TYPE_DISABLED_FREEFALL);
+        }
+
+        obj->Clear_Model_Condition_State(MODELCONDITION_FREEFALL);
+    }
+
+    if (data->m_killWhenRestingOnGround && !above && Is_Low_Velocity(m_vel)
+        && (!obj->Is_KindOf(KINDOF_DRONE) || obj->Is_Effectively_Dead()
```
Comment (tomsons26) [inline src/game/logic/object/update/physicsupdate.cpp]: split this up so its easier to read

File: src/game/logic/object/update/physicsupdate.cpp (hunk 5 above)
Comment (jonwil) [reply src/game/logic/object/update/physicsupdate.cpp]: fixed.

File: src/game/logic/object/update/physicsupdate.cpp [hunk 7]
```diff
@@ -13,17 +13,452 @@
...
+    m_fallHeightDamageFactor(1.0f),
+    m_pitchRollYawFactor(2.0f),
+    m_vehicleCrashesIntoBuildingWeapon(g_theWeaponStore->Find_Weapon_Template("VehicleCrashesIntoBuildingWeapon")),
+    m_vehicleCrashesIntoNonBuildingWeapon(g_theWeaponStore->Find_Weapon_Template("VehicleCrashesIntoNonBuildingWeapon"))
+{
+}
+
+void PhysicsBehaviorModuleData::Build_Field_Parse(MultiIniFieldParse &p)
+{
+    static const FieldParse dataFieldParse[] = {
```
Comment (tomsons26) [inline src/game/logic/object/update/physicsupdate.cpp:66]: look at how others like these are formatted and follow that formatting

File: src/game/logic/object/update/physicsupdate.cpp (hunk 6 above)
Comment (jonwil) [reply src/game/logic/object/update/physicsupdate.cpp]: fixed.

File: src/game/logic/object/update/physicsupdate.cpp (hunk 7 above)
Comment (jonwil) [reply src/game/logic/object/update/physicsupdate.cpp:66]: fixed.

**Review by tomsons26 — APPROVED** [review]

**Pull request merged**